addons/account/static/src/js/account_followup_report_widgets.js

```diff
@@ -0,0 +1,168 @@
+        'click .oe-account-set-next-action': 'setNextAction',
+        'click #saveNextAction': 'saveNextAction',
+        'click .oe-account-followup-one-week': 'setNextAction',
+        'click .oe-account-followup-two-weeks': 'setNextAction',
+        'click .oe-account-followup-one-month': 'setNextAction',
+        'click .oe-account-followup-two-months': 'setNextAction',
+    }, openerp.account.ReportWidgets.prototype.events),
+    saveNextAction: function(e) {
+        e.stopPropagation();
+        e.preventDefault();
+        var note = $("#nextActionNote").val().replace(/\r?\n/g, '<br />').replace(/\s+/g, ' ');
+        var target_id = $("#nextActionModal #target_id").val();
+        var date = $("#nextActionDate").val();
+        date = openerp.web.parse_value(date, {type:'date'})
+        var contextModel = new openerp.Model('account.report.context.followup');
+        return contextModel.call('change_next_action', [[parseInt(target_id)], date, note]).then(function (result) {
+            $('#nextActionModal').modal('hide');
+            $('div.page.' + target_id).find('.oe-account-next-action').html(openerp.qweb.render("nextActionDate", {'note': note, 'date': date}));
+        });
+    },
+    setNextAction: function(e) {
+        e.stopPropagation();
+        e.preventDefault();
+        var target_id = $(e.target).parents("div.page").attr("class").split(/\s+/)[3];
+        var dt = new Date();
+        switch($(e.target).attr("class").split(/\s+/)[2]) {
+            case 'oe-account-followup-one-week':
+                dt.setDate(dt.getDate() + 7);
+                break;
+            case 'oe-account-followup-two-weeks':
+                dt.setDate(dt.getDate() + 14);
+                break;
+            case 'oe-account-followup-one-month':
+                dt.setMonth(dt.getMonth() + 1);
+                break;
+            case 'oe-account-followup-two-months':
+                dt.setMonth(dt.getMonth() + 2);
+                break;
+        }
+        $('.oe-account-picker-next-action-date').data("DateTimePicker").setValue(moment(dt));
+        $("#nextActionModal #target_id").val(target_id);
+        $('#nextActionModal').on('hidden.bs.modal', function (e) {
+            $(this).find('form')[0].reset();
+        });
+        $('#nextActionModal').modal('show');
+    },
+    onChangeBlocked: function(e) {
+        e.stopPropagation();
+        e.preventDefault();
+        var checkbox = $(e.target).is(":checked")
+        var target_id = $(e.target).parents('tr').attr("class").split(/\s+/)[1];
+        var model = new openerp.Model('account.move.line');
+        model.call('write', [[parseInt(target_id)], {'blocked': checkbox}])
+    },
+    onKeyPress: function(e) {
+        var report_name = $("div.page").attr("data-report-name");
+        if ((e.which === 13 || e.which === 10) && (e.ctrlKey || e.metaKey) && report_name == 'followup_report') {
+            $('a.btn-primary.followup-email').trigger('click');
```

Comment:
you don't have to use the all path ('a.btn-primary.followup-email') furthermore btn-primary is a class use for style and could be changed any time for redesign purpose. Just using $('.followup-email') would be faster and better as a selector. Even faster and better, you could just call directly the "sendFollowUp" method that is bound on the click event of .followup-email.